### PR TITLE
add gemini-cli

### PR DESCRIPTION
**Pull request by aman-moh**
### Related GitHub Issue

Closes: # 

### Pre-Submission Checklist

- [ ] **Issue Linked**: This PR is linked to an approved GitHub Issue (see "Related GitHub Issue" above).
- [ ] **Scope**: My changes are focused on the linked issue (one major feature/fix per PR).
- [ ] **Self-Review**: I have performed a thorough self-review of my code.
- [ ] **Testing**: New and/or updated tests have been added to cover my changes (if applicable).
- [ ] **Documentation Impact**: I have considered if my changes require documentation updates (see "Documentation Updates" section below).
- [ ] **Contribution Guidelines**: I have read and agree to the [Contributor Guidelines](/CONTRIBUTING.md).

### Get in Touch

----

> [!IMPORTANT]
> Add `gemini-cli` provider with OAuth authentication, update validation, UI components, and i18n support.
> 
>   - **Behavior**:
>     - Adds `gemini-cli` provider using OAuth authentication, no API key needed.
>     - Integrates `gemini-cli` into `validateApiConfiguration()` and `validateModelsAndKeysProvided()` in `validate.ts`.
>   - **UI Components**:
>     - Adds `GeminiCli` component in `providers` for settings UI.
>     - Updates `ApiOptions.tsx` to include `gemini-cli` settings.
>   - **i18n**:
>     - Adds translations for `gemini-cli` related strings in multiple locales including `tr`, `vi`, `zh-CN`, and `zh-TW`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for 09cb66d787ac0a0677de9b5f53868cacf244813e. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: src/api/providers/__tests__/gemini-cli.spec.ts [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+		it("should fall back to base provider implementation", async () => {
+			const content = [{ type: "text" as const, text: "Hello world" }]
+			const tokenCount = await handler.countTokens(content)
+
+			// Should return a number (tiktoken fallback)
+			expect(typeof tokenCount).toBe("number")
+			expect(tokenCount).toBeGreaterThan(0)
+		})
+	})
+})
```
Comment (ghost) [inline src/api/providers/__tests__/gemini-cli.spec.ts:329]: Consider adding test cases for:
- Network failures during OAuth token refresh
- Malformed OAuth credential files (invalid JSON)
- Timeout scenarios during project discovery

These edge cases would improve the robustness of the test suite.

File: src/api/providers/gemini-cli.ts [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+						yield parsed
+					} catch (e) {
+						console.error("Error parsing SSE data:", e)
+					}
+				}
+			}
+		}
+	}
+
+	async *createMessage(
```
Comment (ghost) [inline src/api/providers/gemini-cli.ts:230]: Consider adding JSDoc comments to document the public methods, especially  and . This would help other developers understand the expected parameters and return values.

**Comment by daniel-lxs** [comment]
Hey @aman-moh We removed gemini-cli a while back, sorry!

**Comment by tolgakaratas** [comment]
guys you should find a way for support gemini-cli. It could be super

> Hey @aman-moh We removed gemini-cli a while back, sorry!

guys you should find a way for support gemini-cli. It could be super

**Comment by aman-moh** [comment]
@tolgakaratas im not sure if it's allowed but you can actually use roo code with gemini cli using this - https://github.com/aman-moh/Roo-Code-Gemini-CLI
